### PR TITLE
Fix bug about metadata.json, we should check the file name first, ins…

### PR DESCRIPTION
**Pull request by wangzhen11aaa**
Hi, We should check whether does  "metadata.json" exist in file path first.

### CODE REVIEW

File: pkg/new_storage/general.go [hunk 1]
```diff
@@ -88,6 +88,17 @@ func isLegacyBackup(backupName string) (bool, string, string) {
 	return false, backupName, ""
 }
 
+func getMetadataFilePath(file RemoteFile) string {
+	_metadataFilePath := ""
```
Comment (Slach) [inline pkg/new_storage/general.go:92]: this is bad code style
please remove underscore

File: pkg/new_storage/general.go [hunk 2]
```diff
@@ -135,7 +147,7 @@ func (bd *BackupDestination) BackupList() ([]Backup, error) {
 		if err != nil {
 			result = append(result, Backup{
 				metadata.BackupMetadata{
-					BackupName: strings.Trim(o.Name(), "/"),
+					BackupName: metadataFilePath,
```
Comment (Slach) [inline pkg/new_storage/general.go:150]: looks like you broke 
`clickhouse-backup list remote`
for other storage types
you should set `backup_name` here
not `backup_name/metadata.json`

Looks like you previous fixes with delimiter and recursive was wrong
could you clarify why `Bucket.Get` from COS on `cos.go`  Walk with delimiter=/ for !recursive was not work properly?








functionality

File: pkg/new_storage/general.go [hunk 3]
```diff
@@ -151,7 +163,7 @@ func (bd *BackupDestination) BackupList() ([]Backup, error) {
 		if err := json.Unmarshal(b, &m); err != nil {
 			result = append(result, Backup{
 				metadata.BackupMetadata{
-					BackupName: strings.Trim(o.Name(), "/"),
+					BackupName: metadataFilePath,
```
Comment (Slach) [inline pkg/new_storage/general.go:166]: looks like you broke 
`clickhouse-backup list remote`
for other storage types
you should set `backup_name` here
not `backup_name/metadata.json`

**Review by Slach — CHANGES_REQUESTED** [review]

File: pkg/new_storage/general.go (hunk 3 above)
Comment (wangzhen11aaa) [reply pkg/new_storage/general.go:166]: yes, I will check that. thanks.

File: pkg/new_storage/general.go (hunk 2 above)
Comment (wangzhen11aaa) [reply pkg/new_storage/general.go:150]: > looks like you broke `clickhouse-backup list remote` for other storage types you should set `backup_name` here not `backup_name/metadata.json`
> 
> Looks like you previous fixes with delimiter and recursive was wrong could you clarify why `Bucket.Get` from COS on `cos.go` Walk with delimiter=/ for !recursive was not work properly?
> 
> functionality

Yes

**Pull request closed without merging**